File: src/main/java/com/example/probation/core/dto/CreateUserDto.java
package com.example.probation.core.dto;

import com.example.probation.validatiion.annotation.Password;
import com.example.probation.validatiion.annotation.UniqueEmail;
import com.example.probation.validatiion.annotation.UniqueUsername;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Size;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CreateUserDto {
    @UniqueUsername
    @NotBlank(message = "{name.notempty}")
    @Size(min = 3, max = 32, message = "{name.size}")
    private String username;
    @NotBlank(message = "{description.notempty}")
    @Size(min = 1, max = 1000, message = "{description.size}")
    private String description;
    @NotBlank(message = "{password.notempty}")
    @Password
    private String password;
    @Email
    @NotBlank(message = "{email.notempty}")
    @UniqueEmail
    private String email;
}

File: src/main/kotlin/com/example/probation/core/dto/CreateUserDto.kt
package com.example.probation.core.dto

import com.example.probation.validation.annotation.Password
import com.example.probation.validation.annotation.UniqueEmail
import com.example.probation.validation.annotation.UniqueUsername

import javax.validation.constraints.Email
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotEmpty
import javax.validation.constraints.Size

data class CreateUserDto(
    @field:UniqueUsername
    @field:NotEmpty(message = "{name.notempty}")
    @field:Size(min = 3, max = 32, message = "{name.size}")
    var username: String? = null,
    @field:NotEmpty(message = "{description.notempty}")
    @field:Size(min = 1, max = 1000, message = "{description.size}")
    var description: String? = null,
    @field:NotBlank(message = "{password.notempty}")
    @field:Password
    var password: String? = null,
    @field:Email
    @field:NotBlank(message = "{email.notempty}")
    @field:UniqueEmail
    var email: String? = null
)
